### PR TITLE
Task/69 implement kick command

### LINKED ISSUE #69: [TASK] Implement KICK command
## Context
`KickCommand.cpp` on `test/command-parsing` is a validated stub: it enforces
exactly 2 arguments, validates the channel name via `requireChannel` and the
target nick via `requireWord`, then logs to stdout. The full IRC logic —
operator permission check, membership verification, removal, and KICK
broadcast — is not yet wired pending Server/Client/Channel context objects.

## Objective
Complete `KickCommand::execute()` with full IRC-compliant behaviour: verify
operator privileges, check membership of both sender and target, remove the
target from the channel, and broadcast the KICK message.

## Acceptation
- [ ] `KickCommand::NAME` is defined as `"KICK"` in the `.cpp`
- [ ] Argument validation (already in stub, must be kept):
  - Exactly 2 arguments (`<channel> <nick>`); optionally a 3rd trailing
    `<reason>` — update to `requireArgsNum(args, 2, 3, ...)` to allow it
  - `args[0]` validated via `requireChannel`
  - `args[1]` validated via `requireWord`
  - `args[2]` (optional reason) validated via `optionalStr`
- [ ] Client must be fully registered
- [ ] Channel must exist; reply `ERR_NOSUCHCHANNEL (403)` if not
- [ ] Sender must be a member of the channel; reply `ERR_NOTONCHANNEL (442)`
  if not
- [ ] Sender must be a channel operator; reply `ERR_CHANOPRIVSNEEDED (482)`
  if not
- [ ] Target nick must be a member of the channel; reply
  `ERR_USERNOTINCHANNEL (441)` if not
- [ ] On success:
  - Broadcast `:<kicker>!<user>@<host> KICK <channel> <target> :<reason>`
    to all channel members (including the target, before removal)
  - Remove target from the channel member list and operator set
- [ ] Compiles cleanly with `-Wall -Wextra -Werror -std=c++98`

## Dependencies
- #60 — `ACommand` validators in place
- #67 — JOIN must be working (clients must be in channels to be kicked)
- Server/Client/Channel context objects must expose: channel map, operator
  set check, member list, member removal, reply sender

## Technical notes
- Current stub enforces exactly 2 args — update to allow the optional reason
  parameter (`2` to `3`) to be RFC-compliant
- Default reason if omitted: the kicker's nick, or a fixed string like
  `"No reason given"`
- `ERR_NOSUCHCHANNEL (403)`: `":server 403 <nick> <channel> :No such channel"`
- `ERR_NOTONCHANNEL (442)`: `":server 442 <nick> <channel> :You're not on that channel"`
- `ERR_CHANOPRIVSNEEDED (482)`: `":server 482 <nick> <channel> :You're not channel operator"`
- `ERR_USERNOTINCHANNEL (441)`: `":server 441 <nick> <target> <channel> :They aren't on that channel"`
- The KICK broadcast must be sent **before** the target is removed from the
  member list, so the target receives the message on their own connection
- If the kicked client is the last operator, the channel may be left without
  an operator — this is acceptable per spec (no auto-promotion required)

### PR DESCRIPTION
**Pull request by lcalero42**
## Summary
Adds `KICK` command implementation with **full parsing and error checking.**

## Why
`KICK` command is needed by the subject in order to **be able to remove by force** someone in a channel

## Changes
- Implemented command parsing
- Implemented command broadcast logic
- Implemented edge cases handling

## Tests
- [x] test with `irssi` with two clients running trying to kick another
- [x] test with `irssi` kicking someone not being an operator
- [x] test with `irssi` by filling a non-existing channel
- [x] test with `irssi` by filling a not joined channel
- [x] test with `irssi` kicking a non existing user in the channel
- [x] test with and without comment optional last parameter (should reply a default comment if you don't fill one)
- [x] Code compiles with `-Wall -Wextra -Werror -std=c++98`, no warnings
- [x] Manual test with `nc -C 127.0.0.1 6667` when relevant
- [x] Manual test with the reference IRC client when relevant
- [x] Edge cases tested (partial messages, malformed commands, client disconnects)

## Linked issue
Closes #69

---

## Merge policy

- Use squash or rebase merges only (no merge commits).
- Ensure dependancy are merged and there's no `blocked` label
- Do not merge if any CI check is red.
- Do not merge without at least one review approval.
- Ensure the branch is up to date with main before merging.

### CODE REVIEW

File: src/commands/operator/KickCommand.cpp [hunk 1]
```diff
@@ -3,11 +3,69 @@
...
+	if (!chan)
+	{
+		ServerReply::reply(client, ServerReply::ERR_NOSUCHCHANNEL, args[0]);
+		return;
+	}
+	if (!target)
+		return;
+	if (target->getNickname() == client.getNickname())
+		return;
+	if (!isValidKick(client, target, chan))
```
Comment (ketodin) [inline src/commands/operator/KickCommand.cpp]: Avoid passing as pointer values like this.
Prefer de-referencing in the function call like `isValidKick(client, *target, *chan)` so you can have instead `const` references parameters with `const type&`. It's usually the best practice to not use pointer if you don't need to know if they're `NULL`. Which is the case here, since you check both `chan` and `target` are indeed not `NULL`

**Review by ketodin — CHANGES_REQUESTED** [review]

File: src/commands/channel/JoinCommand.cpp [hunk 2]
```diff
@@ -45,20 +45,21 @@ JoinCommand::checkAccess(const Client&					 client,
...
+	if (chan.getInviteMode() && !chan.isInvited(client))
 	{
 		ServerReply::reply(client, chan, ServerReply::ERR_INVITEONLYCHAN);
 		return (false);
 	}
-	if (chan.hasMode('k'))
+	if (!chan.isValidKey(""))
 	{
-		if (args.size() < 2 || !chan.isValidKey(args[1]))
+		if ((args.size() < 2) || (!chan.isValidKey(args[1])))
```
Comment (ketodin) [inline src/commands/channel/JoinCommand.cpp]: It is indeed better to have the compare part of the if be guarded between parenthesis. But the second one is not necessary.
`((a < b) || (!c))` should instead be `((a < b) || !c)`

**Review by ketodin — CHANGES_REQUESTED** [review]

**Comment by ketodin** [comment]
Other than the two change requests, the code is clean and the `JOIN` issue is indeed fixed. `KICK` command works properly.

**Review by eterrale — APPROVED** [review]

**Review by ketodin — APPROVED** [review]
